ngraph/core/reference/src/runtime/opt_kernel/timer.cpp

```diff
@@ -0,0 +1,128 @@
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+namespace ngraph
+{
+    namespace perf
+    {
+        Timer::Timer(std::string timer_name)
+            : m_name{std::move(timer_name)}
+        {
+            m_time_blocks.emplace_back("total");
+            m_time_blocks.front().start();
+        }
+
+        void Timer::finish()
+        {
+            // end the total time measurement for this timer
+            m_time_blocks.front().end();
+
+            std::stringstream total_duration;
+            total_duration << std::setw(10) << std::fixed << m_time_blocks.front().elapsed_time()
+                           << " [s]";
+
+            std::vector<std::string> output;
+            if (m_time_blocks.size() == 1)
+            {
+                const std::string timer_name =
+                    "  Total execution time for timer \"" + m_name + "\": " + total_duration.str();
+                const std::string separator_line(timer_name.length() + 2, '=');
+                output.push_back(separator_line);
+                output.push_back(timer_name);
+                output.push_back(separator_line);
+            }
+            else
+            {
+                const std::string timer_name = "  Execution times for timer: " + m_name + "  ";
+                const std::string separator_line(timer_name.length(), '=');
+                output.push_back(separator_line);
+                output.push_back(timer_name);
+                output.push_back(separator_line);
+
+                const auto max_len_block =
+                    std::max_element(std::next(m_time_blocks.begin()),
+                                     m_time_blocks.end(),
+                                     [](const TimeBlock& tb1, const TimeBlock& tb2) {
+                                         return tb1.name().length() < tb2.name().length();
+                                     });
+                const auto longest_name = max_len_block->name().length();
+
+                const auto block_to_str = [&longest_name](const TimeBlock& block) {
+                    std::stringstream stringifier;
+                    const std::string padding(longest_name - block.name().length() + 2, ' ');
+                    stringifier << "  " << block.name() << padding << std::setw(10) << std::fixed
+                                << block.elapsed_time() << " [s]";
+                    return stringifier.str();
+                };
+
+                std::transform(std::next(m_time_blocks.cbegin()),
+                               m_time_blocks.cend(),
+                               std::back_inserter(output),
+                               block_to_str);
+
+                output.push_back(separator_line);
+                output.push_back("  Total time: " + total_duration.str());
+                output.push_back(separator_line);
+            }
+
+            for (const auto& line : output)
+            {
+                std::cout << line << std::endl;
```

Comment:
`std::endl` in loop is not good idea if you don't want to flush stream each time.